ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/RefreshPathAction.java

```diff
@@ -30,18 +36,29 @@
  * @author Vlad Zhukovskiy
  */
 @Singleton
-public class RefreshPathAction extends AbstractPerspectiveAction {
+public class RefreshPathAction extends AbstractPerspectiveAction implements ActivePartChangedHandler {
 
     private final AppContext appContext;
 
+    private PartPresenter activePart;
+
     @Inject
-    public RefreshPathAction(AppContext appContext) {
-        super(singletonList(PROJECT_PERSPECTIVE_ID), "Refresh", "Refresh path", null, null);
+    public RefreshPathAction(AppContext appContext,
+                             CoreLocalizationConstant localizationConstant,
+                             EventBus eventBus) {
+        super(singletonList(PROJECT_PERSPECTIVE_ID), localizationConstant.refreshActionTitle(), localizationConstant.refreshActionDescription());
         this.appContext = appContext;
+
+        eventBus.addHandler(ActivePartChangedEvent.TYPE, this);
```

Comment:
The action is displayed as part menu item and has to be displayed only for Project explorer.